spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractJdbcAggregateJobQueryDaoTests.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.batch;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.repository.dao.AbstractJdbcBatchMetadataDao;
+import org.springframework.batch.item.database.support.DataFieldMaxValueIncrementerFactory;
+import org.springframework.cloud.dataflow.core.database.support.DatabaseType;
+import org.springframework.cloud.dataflow.core.database.support.MultiSchemaIncrementerFactory;
+import org.springframework.cloud.dataflow.schema.AppBootSchemaVersion;
+import org.springframework.cloud.dataflow.schema.service.impl.DefaultSchemaService;
+import org.springframework.cloud.dataflow.server.repository.JdbcAggregateJobQueryDao;
+import org.springframework.cloud.dataflow.server.service.JobServiceContainer;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+abstract class AbstractJdbcAggregateJobQueryDaoTests extends AbstractDaoTests {
+
+	private static final String BASE_JOB_INST_NAME = "JOB_INST_";
+
+	public JdbcSearchableJobInstanceDao jdbcSearchableJobInstanceDao;
+
+	@Mock
+	private JobServiceContainer jobServiceContainer;
+
+	private JdbcAggregateJobQueryDao jdbcAggregateJobQueryDao;
+
+	private DataFieldMaxValueIncrementerFactory incrementerFactory;
+
+	private DatabaseType databaseType;
+
+	protected void prepareForTest(JdbcDatabaseContainer dbContainer,  String schemaName, DatabaseType databaseType) throws Exception {
+		super.prepareForTest(dbContainer, schemaName);
+		this.jdbcAggregateJobQueryDao = new JdbcAggregateJobQueryDao(super.getDataSource(), new DefaultSchemaService(),
+			this.jobServiceContainer);
+		jdbcSearchableJobInstanceDao = new JdbcSearchableJobInstanceDao();
+		jdbcSearchableJobInstanceDao.setJdbcTemplate(new JdbcTemplate(super.getDataSource()));
+		incrementerFactory = new MultiSchemaIncrementerFactory(super.getDataSource());
+		this.databaseType = databaseType;
+	}
+
+	@Test
+	void getJobInstancesForBoot3AndBoot2Instances() throws Exception {
+		assertThat(this.jdbcAggregateJobQueryDao.getJobInstance(1, "boot2")).isNull();
```

Comment:
Another example of how could clear up the API, this would become
```java
assertThat(this.jdbcAggregateJobQueryDao.getJobInstance(1,  AppBootSchemaVersion.BOOT2)).isNull();
```

Reply:
Agreed.   Will resolve this in #5533